File: src/solution_java/easy/_104MaximumDepthOfBinaryTree.java
package solution_java.easy;

import solution_java.TreeNode;

public class _104MaximumDepthOfBinaryTree {
    public class Solution {
        public int maxDepth(TreeNode root) {
            if (root == null) {
                return 0;
            }
            return nodeDepth(root);
        }

        private int nodeDepth(TreeNode root) {
            if (root == null) {
                return 0;
            }
            int childDepth = Math.max(nodeDepth(root.left), nodeDepth(root.right));
            return childDepth + 1;
        }
    }
}

File: src/easy/_104MaximumDepthOfBinaryTree.kt
package easy

import TreeNode

class _104MaximumDepthOfBinaryTree {
    class Solution {
        fun maxDepth(root: TreeNode?): Int {
            if (root == null) {
                return 0
            }
            return travelTree(root)
        }

        private fun travelTree(node: TreeNode?): Int {
            if (node == null) {
                return 0
            }
            val childDepth = Math.max(travelTree(node.left), travelTree(node.right))
            return childDepth + 1
        }
    }
}